yang/frr-nexthop.yang

```diff
@@ -100,110 +105,143 @@ module frr-nexthop {
+        type frr-nexthop:gateway-address;
+        description
+          "The nexthop gateway address.";
+      }
+      leaf interface {
+        type frr-interface:interface-ref;
+        description
+          "The nexthop egress interface.";
+      }
+      leaf onlink {
+        type boolean;
+        default false;
+        description
+          "Nexthop is directly connected.";
       }
-      description
-        "The nexthop vrf name, if different from the route.";
     }
 
     leaf interface {
+      when "../nh-type = 'ifindex'";
       type frr-interface:interface-ref;
       description
         "The nexthop egress interface.";
     }
-
     leaf bh-type {
+      when "../nh-type = 'blackhole'";
       type blackhole-type;
       description
         "A blackhole sub-type, if the nexthop is a blackhole type.";
     }
-
     leaf flags {
       type uint32;
       description
         "The nexthop's raw flags value.";
     }
-
-    leaf is-duplicate {
-      type empty;
+    leaf duplicate {
+      type boolean;
+      default false;
       description
         "Duplicate nexthop; will be ignored.";
     }
-
-    leaf is-recursive {
-      type empty;
+    leaf recursive {
+      type boolean;
+      default false;
       description
         "Nexthop must be resolved through another gateway.";
     }
-
-    leaf is-onlink {
-      type empty;
-      description
-        "Nexthop is directly connected.";
-    }
-
-    leaf is-active {
-      type empty;
+    leaf active {
+      type boolean;
+      default true;
       description
         "Nexthop is active.";
     }
-
     uses rt-types:mpls-label-stack {
       description
         "Nexthop's MPLS label stack.";
     }
-
     leaf mtu {
       type uint32;
       description
         "The nexthop's specific MTU.";
     }
   }
 
-  // End of nexthop
-  /*
-   * Nexthop-group container
-   */
 
-  grouping frr-nexthop-group {
+  container frr-nexthop {
     description
-      "A nexthop-group, represented as a list of nexthop objects.";
-    leaf name {
-      type string;
-      description
-        "The nexthop-group name.";
-    }
-
+      "FRR nexthop.";
     list entry {
       key "id";
       description
         "A list of nexthop objects.";
       leaf id {
         type uint32;
         description
-          "Identifies a nexthop within a nexthop group; the entries
+          "Identifies a nexthop within a list of nexthops; the entries
            are ordered by id value, and the value has no other meaning.";
       }
+      uses frr-nexthop-attributes;
+    }
+  }
 
-      uses frr-nexthop;
+  typedef frr-nexthop-ref {
+    type leafref {
+      path "/frr-nexthop:frr-nexthop/frr-nexthop:entry/frr-nexthop:id";
     }
+    description
+      "Reference to a nexthop.";
   }
 
-  // End of frr-nexthop-group
+  container frr-nexthop-group {
+    description
+      "A nexthop-group, represented as a list of nexthop objects.";
+
+    list group-entry {
```

Comment:
So this is where the actual definition of a group begins right?
Could we call this something else besides `group-entry`. Maybe even just `nexthop-group` will work?

Reply:
+1, this is the global list of nexthop groups, `group-entry` isn't descriptive